rmf_site_editor/src/widgets/preferences.rs

```diff
@@ -0,0 +1,106 @@
+use bevy::prelude::*;
+use bevy_egui::{
+    egui::{self, DragValue, Slider},
+    EguiContext,
+};
+use rmf_site_format::{AnchorParams, AssociatedGraphs, Edge, Preferences, SiteProperties};
+
+use crate::site::{should_display_lane, update_lane_visuals, LaneEnds, LaneSegments, SiteAssets};
+pub struct PreferenceEvent;
+#[derive(Debug, Clone, Resource, Default)]
+pub struct PreferencePanel {
+    enable: bool,
+}
+
+fn enable_preference_ui(
+    mut events: EventReader<PreferenceEvent>,
+    mut panel: ResMut<PreferencePanel>,
+) {
+    for event in events.iter() {
+        panel.enable = true;
+    }
+}
+
+struct RedrawLanes;
+
+fn draw_preference_ui(
+    mut panel: ResMut<PreferencePanel>,
+    mut site_properties: Query<&mut SiteProperties>,
+    mut egui_context: ResMut<EguiContext>,
+    mut redraw_lanes: EventWriter<RedrawLanes>,
+) {
+    if !panel.enable {
+        return;
+    }
+
+    let mut properties = site_properties.get_single_mut();
+
+    if let Ok(props) = properties.as_mut() {
+        // If SiteProperties doesn't exist it is probably because it has not loaded
+        if props.preferences.is_none() {
+            props.preferences = Some(Default::default());
+            println!("Setting default value");
+        }
+
+        if let Some(preferences) = props.preferences.as_mut() {
+            egui::Window::new("Preferences").show(egui_context.ctx_mut(), |ui| {
+                ui.horizontal(|ui| {
+                    ui.label("Default Lane Width: ");
+                    let lane_width_entry = DragValue::new(&mut preferences.default_lane_width)
+                        .clamp_range(0.0..=10000.0);
```

Comment:
`0.0` is a bit of a risky lower bound that would make it completely invisible, I'd suggest making it something that is not completely invisible but still covers the lowest common case (maybe 0.1 m?)